lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -73,9 +76,9 @@ def add_key_from_key_content
     path = create_temporary_file(user_id, resource[:key_content])
     command = "gpg --import #{path}"
     begin
-      output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
+    output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
     rescue Puppet::ExecutionFailure => e
-      raise Puppet::Error, "Error while importing key #{resource[:key_id]} using key content:\n#{output}}"
+      raise Puppet::Error, "Error while importing key #{resource[:key_id]} using key content: #{output}"
```

Comment:
If you want to get the reason for the `Puppet::ExecutionFailure` then you can probably check the `e` variable. Try calling `#inspect` on it, then you can work that into the `Puppet::Error` you're raising, thereby helping people figure out why it failed.